orangecontrib/educational/widgets/owgradientdescent.py

```diff
@@ -166,6 +167,17 @@ class OWGradientDescent(OWWidget):
     auto_play_speed = settings.Setting(1)
     stochastic = settings.Setting(False)
 
+    # SGD optimizers
+    _sgd, _momentum, _nag, _adagrad, _rmsprop, _adadelta, _adam, \
+    _adamax = range(8)
```

Comment:
Make this:
```py
( _OPT_SGD, _OPT_MOMENTUM, _OPT_NAG, _OPT_ADAGRAD,
 _OPT_RMSPROP, _OPT_ADADELTA, _OPT_ADAM, _OPT_ADAMAX) = range(8)
```
All-uppercase because they are constants, `_OPT` prefix to namespace them a little. Alternatively,
```py
class _Optimizer:
    SGD, MOMENTUM, NAG, ... = range(8)
```
Then use this as `self._Optimizer.SGD`. Alternatively, and **preferred**, use types (e.g. `SGD`, `Adam`, `Adamax`, ...) imported from sgd_optimizers module directly, i.e.
```py
if self.opt_type == Momentum:
```